setup.sh

```diff
@@ -1,23 +1,46 @@
 #!/bin/bash
 
+# Brief description of the installation dirs:
+#
+# For all installations:
+# - $prefix/bin/kw: kw binary
+# - $prefix/lib/kw/: code files not to be invoked externally (i.e. src)
+# - $prefix/share/kw/: static data such as sounds, etc files and documentation
+# - $HOME/.kw/: user data, such as the configm storage and user-global config
+#   file
+#
+# Only if $prefix == /usr (system-wide installation):
+# - /usr/share/man/man1/kw.1.gz: man page
+# - /usr/share/bash-completion/completions/kw: bash autocompletion
+# - /usr/share/fish/completions/kw.fish: fish completions
+# - /usr/etc/kw/kworkflow.config: config file
+
 set -e
 
-declare -r APPLICATIONNAME="kw"
-declare -r SRCDIR="src"
-declare -r CONFIG_DIR="etc"
-declare -r INSTALLTO="$HOME/.config/$APPLICATIONNAME"
-declare -r KW_DIR="$HOME/$APPLICATIONNAME"
+. src/kwio.sh --source-only
 
-declare -r SOUNDS="sounds"
-declare -r BASH_AUTOCOMPLETE="bash_autocomplete"
-declare -r DOCUMENTATION="documentation"
+declare prefix
+declare system_wide
 
-declare -r FISH_CONFIG_PATH="$HOME/.config/fish"
-declare -r FISH_COMPLETION_PATH="$FISH_CONFIG_PATH/completions"
+## Following are the install paths
+#  Note: we use "dir"-ended variables for paths to directories that only contain
+#  kw files and "path"-ended variables for paths to specific kw files.
 
-declare -r CONFIGS_PATH="configs"
+# Used in all installations
+declare -r binpath="$prefix/bin/kw"
+declare -r libdir="$prefix/lib/kw/"
+declare -r sharedir="$prefix/share/kw/"
```

Comment:
Yeah, they would be hidden (inside `.local`, though), but then I checked my own home and saw that I have `bin`, `lib` and `share` already created by other programs (being Git one of them). So I think it should be fine to drop kw's files there as well.

Reply:
IMHO it is more elegant if we hide these files from users, since there is no standard for this sort of directories in the home dir we could add all the main files from `kw` at `.bin` or maybe create a `.kw` that will keeps everything hidden from the user.

P.s: I don't have this directories in my home.